### ISSUE TITLE
[Issue]: WARNING  ROCm: hsa version detect failed

### ISSUE THREAD

**Issue by Bercraft** [vladmandic/automatic#3475]
### Issue Description

After clean install the dev branch, with xl checkpoints or pony xl, with different vae, with vae upcasting in default, true, false,with clip skip 1, every generation in blank or noise. 
Master branch works.
Second image full quality off.
![00001-oneFORALLPonyFantasy_v20DPO-score_9_score_8_up_score_7_up](https://github.com/user-attachments/assets/2aae57d6-f6a0-4bc0-8bca-38594ea143f3)
![00002-oneFORALLPonyFantasy_v20DPO-score_9_score_8_up_score_7_up](https://github.com/user-attachments/assets/dd50d0e0-5287-4895-821e-599f1b4d4c15)

### Version Platform Description

windows 10
Threadripper 1920x
Radeon w5700 pro

### Relevant log output

```shell
17:17:38-226370 INFO     ROCm: AMD toolkit detected
17:17:39-036348 INFO     ROCm: agents=['gfx1010:xnack-']
17:17:39-039363 INFO     ROCm: version=6.1, using agent gfx1010:xnack-
17:17:39-041364 WARNING  ZLUDA support: experimental
17:17:39-099689 INFO     Using ZLUDA in G:\automatic-dev.zluda
17:17:39-101689 WARNING  ROCm: hsa version detect failed
No ROCm runtime is found, using ROCM_HOME='C:\Program Files\AMD\ROCm\6.1''
```

### Backend

Diffusers

### UI

Standard

### Branch

Dev

### Model

StableDiffusion XL

### Acknowledgements

- [X] I have read the above and searched for existing issues
- [X] I confirm that this is classified correctly and its not an extension issue

**Comment by vladmandic** [comment]
many amd gpus require user to manually set level of rocm capabilities allowed to be used.
sdnext sets it automatically for some, but it doesn't have a database of all possibilities.

you can set it manually by setting env variable `HSA_OVERRIDE_GFX_VERSION`
if you can find what is correct value for `HSA_OVERRIDE_GFX_VERSION` for `gfx1010` we can add it to database.

**Comment by Bercraft** [comment]
Sure i will investigate further and report here

**Comment by Bercraft** [comment]
Its HSA_OVERRIDE_GFX_VERSION="10.1.0" for gfx1010 aka radeon w5700 pro, maybe even 5700 and 5700 xt.
Generated with XL and pony, nncf, vae upcasting false. 
Works fine like in master branch and with the same speed +-10%.
Using custom rocmlibs for my gpu : [https://github.com/likelovewant/ROCmLibs-for-gfx1103-AMD780M-APU](url)
Rocm: 6.1.
Here a sdk builder for unsupported gpu : [https://github.com/lamikr/rocm_sdk_builder](url).

Generation test:
![00005-cyberrealisticXL_v22-masterpiece best quality high quality highres ultra detailed](https://github.com/user-attachments/assets/b16b774c-0178-4539-af92-d0b7dd904059)

Parameters:

Prompt: masterpiece, best quality, high quality, highres, ultra-detailed, baobab, africa, tree, lanscape

Negative: worst quality, low quality, normal quality

Parameters: Steps: 20| Seed: 1159284384| Sampler: DDPM| CFG scale: 6| Size: 1024x1024| Parser: Full parser| Model: cyberrealisticXL_v22| Model hash: 779ed5cd37| VAE: sdxl.fp16.vae| Backend: Diffusers| App: SD.Next| Version: c13b8d2| Operations: txt2img| Pipeline: StableDiffusionXLPipeline| ToDo: 0.3

Saved: 00005-cyberrealisticXL_v22-masterpiece best quality high quality highres ultra detailed.png

**Comment by vladmandic** [comment]
perfect, i've addedd it to installer db so it will autohandled in the future.

**Comment by Bercraft** [comment]
i will reinstall to double check it works.
Thank you Vlad

**Comment by vladmandic** [comment]
note its in dev branch, not master - it will be merged to master in next release.

**Comment by Bercraft** [comment]
Sorry for the late answer, i did a full install even after a pip purge, it took lots of time but everything is working fine.
I do not want to be disrespectful but the message ROCm: hsa version detect failed is quite misleading, maybe its better someting like ROCm: hsa version detect failed, missing gpu database file ?

Thank you as always.

**Comment by vladmandic** [comment]
its not disrespect to have a valid suggestion!

i've changed the message:
> log.warning(f'ROCm: gfx={device.name} could not auto-detect HSA version')

**Comment by RogerWeihrauch** [comment]
Hi All
I hope to be on the correct site of issues with my one.
My Q is on using Automatic by Vlad throwing following warning:
' WARNING  ROCm: device=gfx803 could not auto-detect HSA version '
My env consists of: AMD Ryzen Threadripper 1900, AMDGPU RX580 style GPU, hence the 'gfx803', I think.
Installed ROCm 6.3 successfully on Debian Bookworm.
So:
1) Is it possible to use automatic in general on my RX580 style GPU?
-> Will the env option: ' export HSA_OVERRIDE_GFX_VERSION=gfx803	(for AMDGPU RX580 types)' suffice?
2) if not which things had to be changed/kept in mind for getting it on a RX580?
-> will a more less ROCm version suffice? Ifso, which one would you suggest?
Thank you all for your helpfull responses.
They will be highly appreciated.
Regards,
Roger

**Comment by xhy2008** [comment]
Hi，I‘m trying to build rocblas for my iGPU on Windows,which was not officially supported by HIP SDK.
It was gfx902:xnack-,but I would get an error saying that gfx902:xnack- is not supported while compiling the library.
So I changed the target to gfx900(the most similar one),and still couldn't get sd work.

Using VENV: E:\sdnext\venv
18:46:02-758322 INFO     Starting SD.Next
18:46:02-767687 INFO     Logger: file="E:\sdnext\sdnext.log" level=INFO size=9340 mode=append
18:46:02-770684 INFO     Python: version=3.10.6 platform=Windows bin="E:\sdnext\venv\Scripts\python.exe" venv="E:\sdnext\venv"
18:46:03-386012 INFO     Version: app=sd.next updated=2025-02-04 hash=0dad4de1 branch=master url=https://github.com/vladmandic/sdnext/tree/master ui=main
18:46:03-701380 ERROR    Repository failed to check version: HTTPSConnectionPool(host='api.github.com', port=443): Max retries exceeded with url: /repos/vladmandic/sdnext/branches/master
                         (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate
                         (_ssl.c:997)'))) None
18:46:03-730857 INFO     Platform: arch=AMD64 cpu=AMD64 Family 23 Model 24 Stepping 1, AuthenticAMD system=Windows release=Windows-10-10.0.19045-SP0 python=3.10.6 docker=False
18:46:03-736362 INFO     Args: ['--lowvram', '--use-zluda']
18:46:03-895725 INFO     ROCm: AMD toolkit detected
18:46:05-179564 INFO     ROCm: agents=['gfx902:xnack-']
18:46:05-182563 INFO     ROCm: version=5.7, using agent gfx902:xnack-
18:46:05-184564 INFO     Python: version=3.10.6 platform=Windows bin="E:\sdnext\venv\Scripts\python.exe" venv="E:\sdnext\venv"
18:46:05-548296 INFO     Using ZLUDA in E:\sdnext\.zluda
18:46:05-551299 WARNING  ROCm: device=gfx902:xnack- could not auto-detect HSA version
No ROCm runtime is found, using ROCM_HOME='C:\Program Files\AMD\ROCm\5.7'
18:46:31-708163 INFO     Diffusers update: package=diffusers 0.32.2 current= target=f63d32233f402bd603da8f3aa385aecb9c3d8809
18:47:26-514340 INFO     Install: verifying requirements
18:47:40-335100 INFO     Startup: standard
18:47:40-338101 INFO     Verifying submodules
18:48:36-429462 INFO     Extension installed packages: sd-webui-agent-scheduler ['SQLAlchemy==2.0.37', 'greenlet==3.1.1']
18:48:50-964814 INFO     Extension installed packages: stable-diffusion-webui-rembg ['PyMatting==1.1.13', 'opencv-python-headless==4.11.0.86', 'pooch==1.8.2', 'rembg==2.0.62']
18:48:50-971815 INFO     Extensions enabled: ['Lora', 'sd-extension-chainner', 'sd-extension-system-info', 'sd-webui-agent-scheduler', 'sdnext-modernui', 'stable-diffusion-webui-rembg']
18:48:50-975815 INFO     Install: verifying requirements
18:48:50-992816 INFO     Command line args: ['--lowvram', '--use-zluda'] lowvram=True use_zluda=True args=[]
18:49:05-875594 INFO     Torch: torch==2.2.1+cu118 torchvision==0.17.1+cu118
18:49:05-883596 INFO     Packages: diffusers==0.33.0.dev0 transformers==4.48.2 accelerate==1.3.0 gradio==3.43.2
18:49:08-406438 INFO     Engine: backend=Backend.DIFFUSERS compute=zluda device=cuda attention="Scaled-Dot-Product" mode=no_grad
18:49:08-460441 INFO     Available Styles: folder="models\styles" items=288 time=0.05
Compilation is in progress. Please wait...
Compilation is in progress. Please wait...
Compilation is in progress. Please wait...
Compilation is in progress. Please wait...
18:54:53-270190 INFO     Torch parameters: backend=zluda device=cuda config=Auto dtype=torch.float16 context=no_grad nohalf=False nohalfvae=False upcast=False deterministic=False
                         tunable=[False, False] fp16=pass bf16=fail optimization="Scaled-Dot-Product"
Compilation is in progress. Please wait...

rocBLAS warning: No paths matched C:\Program Files\AMD\ROCm\5.7\bin\\rocblas\library\*gfx902*co. Make sure that ROCBLAS_TENSILE_LIBPATH is set correctly.
Compilation is in progress. Please wait...
19:00:16-438370 WARNING  ZLUDA device failed to pass basic operation test: index=None, device_name=AMD Radeon(TM) Vega 8 Graphics [ZLUDA]
19:00:16-446675 ERROR

Then I would get lots of errors when clicking 'Generate'
Any help would be appreciated.

**Comment by RogerWeihrauch** [comment]
@xhy2008:
Hi. Since I'd also been struggling with ROCm and my RX580 AMDGPU and not really solving my issue:
1) Are you sure your integrated GPU (hence: iGPU) will work with that version of ROCm?
(in my case I got a working ROCm version for my GPU ()gfx802), but This one did not work with any SDNext version/needed torch versions in there)
(pers. remark: even if u will get it working, u will not be really satisfied with doing GenAi on an iGPU: it will be too slow AND become too hot; will be a pitty for that laptop)
2) Sine your error seems to be the same as for me:
' device=gfx902:xnack- could not auto-detect HSA version'; it seems to wanting fall-back to kind of CUDA toolkit:
torch==2.2.1+cu118 torchvision==0.17.1+cu118 (see the ..cu.. mentioning)
3) also there is a problem with any kind of certification:
' certificate verify failed: unable to get local issuer certificate'
4) the env var 'ROCBLAS_TENSILE_LIBPATH' seems not to be set since being requested to be
May be working out those issue will guide you back on track.
Regards,
Roger
PS: I gave up trying to get SDNext running with my RX580 (shows up as gfx802) though; is laying around since then.
